src/actions/receive-profile.js

```diff
@@ -164,6 +168,11 @@ export function finalizeProfileView(
       hasUrlInfo ? getHiddenGlobalTracks(getState()) : null,
       getLegacyHiddenThreads(getState())
     );
+    // Pre-compute which tracks are not available for the active tab.
+    const activeTabHiddenGlobalTracks = computeActiveTabHiddenGlobalTracks(
```

Comment:
Consider changing this to a callback, so that the work can be deferred.

```js
const activeTabHiddenGlobalTracks = () => computeActiveTabHiddenGlobalTracks(
  globalTracks,
  getState() // we need to access per thread selectors inside
);
```